kernel/src/hil/time.rs

```diff
@@ -86,10 +93,26 @@ pub trait Alarm<'a, W = u32>: Time<W> {
     /// let tics = alarm.now().wrapping_add(delta);
     /// alarm.set_alarm(tics);
     /// ```
-    fn set_alarm(&self, tics: W);
+    fn set_alarm(&self, tics: Self::Ticks);
+
+    /// Sets a one-shot alarm to fire when the clock reaches `duration` ticks
+    /// from now.
+    ///
+    /// [`Client#fired`](trait.Client.html#tymethod.fired) is signaled when the
+    /// alarm is elapsed.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// let delta = 1337;
+    /// alarm.set_alarm_from_now(delta);
+    /// ```
+    fn set_alarm_from_now(&self, duration: Self::Ticks) {
```

Comment:
I actually don't think we want to encourage clients to use this kind of helper method. In particular, it's very important for time-related applications/drivers to reason precisely about _when_ `now()` was taken.

I do understand the desire to hide this relatively gross but common line of code.

Perhaps an alternative would be to have `Ticks` implement `core::ops::Add`, `core::ops::Sub` using wrapping logic such that clients can just use `+`, `-`, etc? In the same way `core::num::Wrapping<T>` does.

Reply:
This actually changes the HIL, rather than just supports different counter widths.